### PR TITLE
[pcre2] Explicitly control dependencies

### PR DESCRIPTION
**Pull request by Thomas1664**
**Describe the pull request**

- #### What does your PR fix?
  Add dependencies on `bzip2` and `zlib` to `vcpkg.json`
  Disable searching for optional dependencies

### CODE REVIEW

File: ports/pcre2/deps.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+index 7febf33..50e4c43 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -124,8 +124,8 @@ LIST(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+ INCLUDE_DIRECTORIES(${PROJECT_SOURCE_DIR}/src)
+ 
+ # external packages
+-FIND_PACKAGE( BZip2 )
+-FIND_PACKAGE( ZLIB )
++FIND_PACKAGE( BZip2 REQUIRED )
```
Comment (Osyotr) [inline ports/pcre2/deps.patch:11]: Maybe use `CMAKE_REQUIRE_FIND_PACKAGE_*`?
https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#when-defining-features-explicitly-control-dependencies

Comment (Thomas1664) [reply ports/pcre2/deps.patch:11]: Reminder to self: Push if OpenSSL update gets merged.

**Comment by FrankXie05** [comment]
@Thomas1664 Is the current PR completed?

**Comment by Thomas1664** [comment]
> @Thomas1664 Is the current PR completed?

I think so

**Review by FrankXie05 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**